### PR TITLE
parity with JavaScript

### PR DESCRIPTION
**Pull request by johnataylor**
- add Get function to BotState for parity with JavaScript

this fixes https://github.com/microsoft/botbuilder-js/issues/896

### CODE REVIEW

**Comment by fuselabs** [comment]
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.AI.Luis.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.AI.QnA.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.ApplicationInsights.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.Azure.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.Dialogs.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.TemplateManager.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Configuration.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Connector.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Schema.dll

File: libraries/Microsoft.Bot.Builder/BotState.cs [hunk 1]
```diff
@@ -142,6 +143,23 @@ public virtual async Task DeleteAsync(ITurnContext turnContext, CancellationToke
             await _storage.DeleteAsync(new[] { storageKey }, cancellationToken).ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Returns the raw cached data from the TurnContext.
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder/BotState.cs]: I'm not clear, reading this, aboput the behavior. 

If I poke at the JToken that comes back, am I poking at a clone of the data, or at the underlying data itself? I think documenting the semantics here are worthwhile.

Comment (cleemullins) [reply libraries/Microsoft.Bot.Builder/BotState.cs]: ... and equally important, the semantics of the returned JSON should match the JS SDK Behavior.

Comment (johnataylor) [reply libraries/Microsoft.Bot.Builder/BotState.cs]: What do you mean by "the semantics returned JSON should match the JS SDK Behavior" the shape of the JSON is different - that's the nature of the beast?
Serialization is an antipattern, but it is the pattern adopted in this SDK (against my advice).
Anyhow, academic, this makes writing tings like inspection easier. Its a view on internal state.
Technically the JavaScript is not a copy and this is. There is no such thing as a non-leaky abstraction.

Comment (johnataylor) [reply libraries/Microsoft.Bot.Builder/BotState.cs]: I'll add a comment.
this is now at parity. and the constant "state" is hidden - as it is in the JavaScript

**Comment by fuselabs** [comment]
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.AI.Luis.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.AI.QnA.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.ApplicationInsights.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.Azure.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.Dialogs.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Builder.TemplateManager.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Configuration.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Connector.dll
#### :heavy_check_mark: No Binary Compatibility issues for Microsoft.Bot.Schema.dll

**Pull request merged**